include/ibamr/ConvectiveOperator.h

```diff
@@ -84,6 +84,11 @@ class ConvectiveOperator : public IBTK::GeneralOperator
      */
     virtual void applyConvectiveOperator(int Q_idx, int N_idx) = 0;
 
+    /*!
+     * \brief Compute N = div(Q1 Q2 u).
+     */
+    virtual void applyConvectiveOperator(int Q_idx, int P_idx, int N_idx);
```

Comment:
For smooth things maybe yes, but it is not a good idea for sharp contrast to multiply Q and P. Physically speaking, the right thing to do is (which is what we do) is to have a mass flux $\rho \bf{u}$ computed from phase advection, apply limiter on $T$. For $C_p$ it is not clear whether it should be limited or not, because the quantity being advected is $T$ and not $C_p$. I think we tried both ways---limiting $C_p$ and just interpolation---and we have settled on one versus the other, don't recall which one. @Raamkrishnan ?